src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala

```diff
@@ -110,54 +110,111 @@ private[spark] class FlexibleDataModelCorePropertySyncRelation(
                 p != "startNode" && p != "endNode" && p != "space" && p != "externalId" && p != "type")
         ))
       .toSeq
+    val cursors = if (cursor.nonEmpty) Some(Map("sync" -> cursor)) else None
+    val select = Map("sync" -> SelectExpression(sources = sourceReference))
+    val (futureItemsCursor, executeBackFill) =
+      generateFutureItemsCursor(cursors, instanceType, select).unsafeRunSync()
+
+    Seq(
+      syncOut(
+        futureItemsCursor,
+        cursors = if (executeBackFill) None else cursors,
+        `with` = Map("sync" -> tableExpression),
+        select,
+        selectedColumns
+      ))
+  }
 
-    val instanceSyncRequest = InstanceSyncRequest(
-      `with` = Map("sync" -> tableExpression),
-      cursors = if (cursor.nonEmpty) Some(Map("sync" -> cursor)) else None,
-      select = Map("sync" -> SelectExpression(sources = sourceReference)),
-      includeTyping = Some(true)
-    )
+  /**
+    * Generate future items cursor and validate cursor expiration.
+    * If the cursor has expired, generate a new cursor and do a full back fill.
```

Comment:
What are "future" items? Is it different from what items will be returned when we make a request with cursors?

Reply:
The implementation is now that on the first sync, we do a backfill of data with cursor by id on the query endpoint. Basically list out all data. But before that, we ask the sync endpoint to create a cursor for future items from right now.

Reply:
Is there an api to start sync from now() or now() minus some threshold?

Reply:
But anyway, will take this mental model:
- cursors coming into the datasource and out of it are /sync cursors
- first time run == input cursors aren't set
- on first run we fetch via /query but first stash a /sync cursor that will be returned and used for the next run

Reply:
additionally
- whenever input cursor is expired, it is also going as if it wasn't set at all aka via /query